pkgs/applications/terminal-emulators/kitty/test.patch

```diff
@@ -0,0 +1,32 @@
+diff --git a/kitty_tests/shell_integration.py b/kitty_tests/shell_integration.py
+index b7777d77..b26c64c9 100644
+--- a/kitty_tests/shell_integration.py
++++ b/kitty_tests/shell_integration.py
+@@ -107,7 +107,10 @@ def test_zsh_integration(self):
+                 pty.wait_till(lambda: pty.screen.cursor.shape == CURSOR_BEAM)
+             except TimeoutError as e:
+                 raise AssertionError(f'Cursor was not changed to beam. Screen contents: {repr(pty.screen_contents())}') from e
+-            self.ae(pty.screen_contents(), q)
++            pty.send_cmd_to_child('echo "zri=$ZLE_RPROMPT_INDENT"')
++            pty.wait_till(lambda: pty.screen_contents().count(rps1) == 2)
++            print(pty.screen_contents())
++            self.ae(pty.last_cmd_output(), q)
+             self.ae(pty.callbacks.titlebuf[-1], '~')
+             pty.callbacks.clear()
+             pty.send_cmd_to_child('mkdir test && ls -a')
+diff --git a/shell.nix b/shell.nix
+index 4e5744fd..78c83dd8 100644
+--- a/shell.nix
++++ b/shell.nix
```

Comment:
Can we use the shell at all?

Reply:
No, I was just testing. I want to see the the contents of a variable, see https://github.com/kovidgoyal/kitty/issues/4782. I'll remove the commit and finish this PR once I have my answer.